src/chainstate/burn/db/sortdb.rs

```diff
@@ -3699,6 +3786,52 @@ impl<'a> SortitionHandleTx<'a> {
         Ok(())
     }
 
+    pub fn store_exit_at_reward_cycle_info(
+        &self,
+        exit_at_reward_cycle_info: BlockExitRewardCycleInfo,
+    ) -> Result<(), db_error> {
+        let sql = "INSERT or REPLACE INTO exit_at_reward_cycle_info (block_id, parent_block_id, block_reward_cycle, stacks_block_height_in_cycle, invalid_reward_cycles) VALUES (?, ?, ?, ?, ?)";
+        let args: &[&dyn ToSql] = &[
+            &exit_at_reward_cycle_info.block_id,
+            &exit_at_reward_cycle_info.parent_block_id,
+            &u64_to_sql(exit_at_reward_cycle_info.block_reward_cycle)?,
+            &u64_to_sql(exit_at_reward_cycle_info.stacks_block_height_in_cycle)?,
+            &serde_json::to_string(&exit_at_reward_cycle_info.invalid_reward_cycles).unwrap(),
+        ];
+        self.execute(sql, args)?;
+
+        if let Some(exit_proposal) = exit_at_reward_cycle_info.curr_exit_proposal {
+            let sql =
```

Comment:
Should this verify that exactly one row was set?  Is it possible that this `UPDATE` touches zero rows?

Reply:
The logic right before this inserts a row with that block_id, and it should fail if the insert failed? 

Do you think an assertion is still necessary here?